From 175435c621e895209d983ca6962b3bc77d5e15e3 Mon Sep 17 00:00:00 2001
From: Aditya Goel <agoel4512@gmail.com>
Date: Tue, 7 Nov 2023 11:47:35 +0000
Subject: [PATCH] Link to abseil_dll

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 843da66..dfb1244 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -528,7 +528,7 @@ target_include_directories(onnx PUBLIC
   $<BUILD_INTERFACE:${ONNX_ROOT}>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
   $<INSTALL_INTERFACE:include>)
-target_link_libraries(onnx PUBLIC onnx_proto)
+target_link_libraries(onnx PUBLIC onnx_proto $ENV{PREFIX}/Library/lib/abseil_dll.lib)
 add_onnx_global_defines(onnx)
 
 if(ONNX_BUILD_PYTHON)
