From 18fb86cb2fc45dc134b500da5e5ab2cc4828c383 Mon Sep 17 00:00:00 2001
From: Aditya Goel <agoel4512@gmail.com>
Date: Tue, 7 Nov 2023 11:47:35 +0000
Subject: [PATCH] Link to abseil_dll

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4799557..de44ef2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -464,7 +464,7 @@ endif()
 set_target_properties(onnx PROPERTIES CXX_VISIBILITY_PRESET hidden)
 set_target_properties(onnx PROPERTIES VISIBILITY_INLINES_HIDDEN ON)
 
-target_link_libraries(onnx PUBLIC onnx_proto)
+target_link_libraries(onnx PUBLIC onnx_proto $ENV{PREFIX}/Library/lib/abseil_dll.lib)
 add_onnx_global_defines(onnx)
 
 if(ONNX_BUILD_PYTHON)
