From 4381d246acc48b816065ea54966bb9019ca251a2 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sun, 10 Mar 2024 17:47:05 -0400
Subject: [PATCH 3/5] include cinttypes in cpp2py_export.cc

---
 onnx/cpp2py_export.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/onnx/cpp2py_export.cc b/onnx/cpp2py_export.cc
index bc2594db..55fed8b5 100644
--- a/onnx/cpp2py_export.cc
+++ b/onnx/cpp2py_export.cc
@@ -2,6 +2,8 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+#include <cinttypes>
+
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
 
-- 
2.40.1

