From 8962feadc919b02171be884aaad51091a3f7f186 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sun, 10 Mar 2024 17:57:54 -0400
Subject: [PATCH 5/6] Link protobuf privately

---
 CMakeLists.txt | 30 +++++++++++++++++-------------
 1 file changed, 17 insertions(+), 13 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b4fdf131..d1a4bb78 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -487,34 +487,38 @@ else()
 endif()
 target_compile_definitions(onnx_proto PRIVATE ${ONNX_API_DEFINE})
 
+if(CMAKE_SYSTEM_NAME STREQUAL "AIX")
+  # whole-archive linker option not available on AIX.
+  # So, create a object library
+  add_library(onnx OBJECT ${ONNX_SRCS})
+else()
+  add_library(onnx ${ONNX_SRCS})
+endif()
+
 if(ONNX_USE_LITE_PROTO)
   if(TARGET protobuf::libprotobuf-lite)
-    target_link_libraries(onnx_proto PUBLIC protobuf::libprotobuf-lite PRIVATE ${protobuf_ABSL_USED_TARGETS})
+    target_link_libraries(onnx_proto PRIVATE protobuf::libprotobuf-lite ${protobuf_ABSL_USED_TARGETS})
+    target_link_libraries(onnx PRIVATE protobuf::libprotobuf-lite ${protobuf_ABSL_USED_TARGETS})
   else()
-    target_link_libraries(onnx_proto PUBLIC ${PROTOBUF_LITE_LIBRARIES})
+    target_link_libraries(onnx_proto PRIVATE ${PROTOBUF_LITE_LIBRARIES})
+    target_link_libraries(onnx PRIVATE ${PROTOBUF_LITE_LIBRARIES})
   endif()
 else()
   if(TARGET protobuf::libprotobuf)
-    target_link_libraries(onnx_proto PUBLIC protobuf::libprotobuf PRIVATE ${protobuf_ABSL_USED_TARGETS})
+    target_link_libraries(onnx_proto PRIVATE protobuf::libprotobuf ${protobuf_ABSL_USED_TARGETS})
+    target_link_libraries(onnx PRIVATE protobuf::libprotobuf ${protobuf_ABSL_USED_TARGETS})
   else()
-    target_link_libraries(onnx_proto PUBLIC ${PROTOBUF_LIBRARIES})
+    target_link_libraries(onnx_proto PRIVATE ${PROTOBUF_LIBRARIES})
+    target_link_libraries(onnx PRIVATE ${PROTOBUF_LIBRARIES})
   endif()
 endif()
 add_onnx_global_defines(onnx_proto)
 
-if(CMAKE_SYSTEM_NAME STREQUAL "AIX")
-  # whole-archive linker option not available on AIX.
-  # So, create a object library
-  add_library(onnx OBJECT ${ONNX_SRCS})
-else()
-  add_library(onnx ${ONNX_SRCS})
-endif()
-
 target_include_directories(onnx PUBLIC
   $<BUILD_INTERFACE:${ONNX_ROOT}>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
   $<INSTALL_INTERFACE:include>)
-target_link_libraries(onnx PUBLIC onnx_proto $ENV{PREFIX}/Library/lib/abseil_dll.lib)
+target_link_libraries(onnx PRIVATE onnx_proto)
 add_onnx_global_defines(onnx)
 
 if(BUILD_ONNX_PYTHON)
-- 
2.40.1

